core/src/test/java/brooklyn/util/crypto/SecureKeysAndSignerTest.java

```diff
@@ -93,6 +95,40 @@ public void testReadRsaKey() throws Exception {
         checkNonTrivial(key);
     }
 
+    @Test(expectedExceptions=IllegalStateException.class)
+    public void testReadRsaPublicKeyAsPemFails() throws Exception {
+        // should fail; see next test
+        SecureKeys.readPem(ResourceUtils.create(this).getResourceFromUrl("classpath://brooklyn/util/crypto/sample_rsa.pem.pub"), null);
+    }
+    
+    @Test
+    public void testReadRsaPublicKeyAsAuthKeysWorks() throws Exception {
+        PublicKey key = AuthorizedKeysParser.decodePublicKey(
+            ResourceUtils.create(this).getResourceAsString("classpath://brooklyn/util/crypto/sample_rsa.pem.pub"));
+        KeyPair fromPem = SecureKeys.readPem(ResourceUtils.create(this).getResourceFromUrl("classpath://brooklyn/util/crypto/sample_rsa.pem"), null);        
```

Comment:
Not particularly related to this PR, but on first impressions I'm confused why `brooklyn/util/crypto/sample_rsa.pem` is excluded from RAT check. In the root `pom.xml`, it has `<exclude>**/*.pem.pub</exclude>` but nothing for `*.pem`.